### DISCUSSION TITLE
Allow specifying an existing Floating IP

### DISCUSSION THREAD

**Discussion by amplitudesxd** [mysticaltech/terraform-hcloud-kube-hetzner#1385] [Ideas]
### Description

We want to use an egress node in our Kubernetes cluster, but occasionally need to drain the node pool, which results in the loss of the Floating IP.

We use this IP in the firewalls and whitelists of external services that our cluster connects to.

### Request

Please allow the specification of an existing Floating IP ID for a node. This would help us avoid losing the IP during maintenance, and reduce the operational overhead when this occurs.

**Comment by WebSpider** [comment]
I never actually looked at the egress gateway, but it's documentation says there's more advanced methods of selecting an exit, rather than "just" the IP address.

Looking at the way this terraform module is set up, it would make sense to either:

- Select an existing floating ip, either by IP, label or ID
- Create our own floating IP if absent, if desired to do so (selectable by variable)

**Comment by mysticaltech** [comment]
@amplitudesxd It sounds like a good idea that could be useful for quite a few people, we will look into it, but in the meantime PRs are most welcome.

**Comment by amplitudesxd** [comment]
I'm afraid I don't think I'm quite experienced enough with Terraform to make a PR, though thank you for taking a look!
